perf(common): remove unused methods from DomAdapter

crisbeto opened the pull request:
The `DomAdapter` is present in all Angular apps and its methods aren't tree shakeable. These changes remove the methods that either aren't being used anymore or were only used by our own tests. Note that these changes aren't breaking, because the adapter is an internal API.

The following methods were removed:
* `getProperty` - only used within our own tests.
* `log` - Guaranteed to be defined on `console`.
* `logGroup` and `logGroupEnd` - Only used in one place. It was placed in the `DomAdapter` for built-in null checking.
* `performanceNow` - Only used in one place that has to be invoked through the browser console.
* `supportsCookies` - Unused.
* `getCookie` - Unused.
* `getLocation` and `getHistory` - Only used in one place which appears to have access to the DOM already, because it had direct accesses to `window`. Furthermore, even if this was being used in a non-browser context already, the `DominoAdapter` was set up to throw an error.

The following APIs were changed to be more compact:
* `supportsDOMEvents` - Changed to a readonly property.
* `remove` - No longer returns the removed node.

itea-dev commented:
There is a minor typo in the description.

  > Only used in one place that has to be invoked through the **broser** console.

crisbeto commented on packages/core/test/linker/security_integration_spec.ts:
```diff
@@ -171,41 +167,37 @@ function declareTests(config?: {useJit: boolean}) {
         fixture.detectChanges();
-        // In the browser, reading href returns an absolute URL. On the server side,
-        // it just echoes back the property.
-        let value = isAttribute ? e.getAttribute('href') : getDOM().getProperty(e, 'href');
```
The `DominoAdapter` had a special case for property reads of `href` that turned them into `getAttribute('href')` so this check wasn't actually doing anything.

jelbourn reviewed (approved):
LGTM

Reviewed-for: size-tracking

jelbourn reviewed (approved):
LGTM

Reviewed-for: fw-security

alfaproject commented on packages/platform-browser/src/browser/browser_adapter.ts:
```diff
@@ -121,40 +79,19 @@ export class BrowserDomAdapter extends GenericBrowserDomAdapter {
-    }
-  }
-  return baseElement.getAttribute('href');
+  baseElement = baseElement || document.querySelector('base');
```
```suggestion
  baseElement ||= document.querySelector('base');
```

alfaproject commented on packages/platform-server/src/domino_adapter.ts:
```diff
@@ -107,12 +77,8 @@ export class DominoAdapter extends BrowserDomAdapter {
-    }
     // TODO(alxhub): Need relative path logic from BrowserDomAdapter here?
-    return href;
+    return base ? base.getAttribute('href')! : '';
```
Shouldn't this function signature and name mirror the platform-browser one?

AndrewKushnir commented:
[Presubmit](http://test/OCL:361826200:BASE:361826233:1615310617507:177b7aac).

AndrewKushnir commented:
@crisbeto running tests in Google's codebase indicated that this change breaks a lot of targets (85%) due to the usage of removed methods in older versions of FlexLayout and deprecated `@angular/http` module. This would require some refactoring in Google's codebase (to avoid using removed methods) before landing the change.

crisbeto commented:
`getCookie` has been restored with a comment about why it's there.

AndrewKushnir commented:
@crisbeto FYI we've performed the necessary refactoring in Google's codebase to get it ready for this change. I will run more broad set of tests tonight to see if there are other places that'd require a cleanup and will keep you updated. Thank you.

AndrewKushnir commented:
Btw, it looks like this PR also needs the `fw-platform-server` group approval, so may be @alxhub or @kyliau can have a look at these changes too when they get a chance? Thank you.

kyliau commented on packages/platform-server/src/domino_adapter.ts:
```diff
@@ -105,14 +75,9 @@ export class DominoAdapter extends BrowserDomAdapter {
-    if (base) {
-      href = base.getAttribute('href')!;
-    }
+  getBaseHref(doc: Document): string|null {
```
This will never return null, will it?

alfaproject replied:
But it should return null like the platform browser does. Currently this is not mirroring the behaviour

crisbeto replied:
Changing the behavior now could be considered a breaking change.

kyliau replied:
SGTM, thanks for the update!

AndrewKushnir commented:
[Global Presubmit](https://test.corp.google.com/OCL:361826200:BASE:361952602:1615346969414:77520aec).

crisbeto committed:
perf(common): remove unused methods from DomAdapter

The `DomAdapter` is present in all Angular apps and its methods aren't tree shakeable.
These changes remove the methods that either aren't being used anymore or were only
used by our own tests. Note that these changes aren't breaking, because the adapter
is an internal API.

The following methods were removed:
* `getProperty` - only used within our own tests.
* `log` - Guaranteed to be defined on `console`.
* `logGroup` and `logGroupEnd` - Only used in one place. It was in the DomAdapter for built-in null checking.
* `logGroupEnd` - Only used in one place. It was placed in the DomAdapter for built in null checking.
* `performanceNow` - Only used in one place that has to be invoked through the browser console.
* `supportsCookies` - Unused.
* `getCookie` - Unused.
* `getLocation` and `getHistory` - Only used in one place which appears to have access to the DOM
already, because it had direct accesses to `window`. Furthermore, even if this was being used
in a non-browser context already, the `DominoAdapter` was set up to throw an error.

The following APIs were changed to be more compact:
* `supportsDOMEvents` - Changed to a readonly property.
* `remove` - No longer returns the removed node.

AndrewKushnir commented:
FYI, Global Presubmit is successful. I've started a [new presubmit](https://test.corp.google.com/ui#id=OCL:361826200:BASE:362087011:1615401488080:c39170e0) with a subset of targets to verify that the most recent changes are ok too and once presubmit is done, this PR should be ready to go.

AndrewKushnir commented:
Note to Caretaker: PullApprove still wants the `fw-security` group review, even though Jeremy already approved this PR on behalf of that group in https://github.com/angular/angular/pull/41102#pullrequestreview-606642044. This PR should be ready to go once final presubmits are completed.

AndrewKushnir commented:
FYI, discussed with @crisbeto that this PR should go into master only to avoid any potential breakages in patch versions.
